hail_search/queries/base.py

```diff
@@ -299,63 +316,98 @@ def _parse_sample_data(self, sample_data):
+        In the variant search control flow, project_samples looks like this:
+            {<project_guid>: {<sample_type>: {<family_guid>: [<sample_data>, <sample_data>, ...]}, <sample_type_2>: {<family_guid_2>: []} ...}, <project_guid_2>: ...}
+        """
+        entries_hts = self._load_prefiltered_project_hts(project_samples, n_partitions, **kwargs)
         filtered_project_hts = []
         filtered_comp_het_project_hts = []
+        for ht, project_families in entries_hts:
+            ht, comp_het_ht = self._filter_single_entries_table(ht, project_families, **kwargs)
+            if ht is not None:
+                filtered_project_hts.append(ht)
+            if comp_het_ht is not None:
+                filtered_comp_het_project_hts.append(comp_het_ht)
+
+        return self._merge_filtered_hts(filtered_comp_het_project_hts, filtered_project_hts, n_partitions)
+
+    def _merge_filtered_hts(self, filtered_comp_het_project_hts, filtered_project_hts, n_partitions):
+        ht = self._merge_project_hts(filtered_project_hts, n_partitions)
+        comp_het_ht = self._merge_project_hts(filtered_comp_het_project_hts, n_partitions)
+        return ht, comp_het_ht
+
+    def _load_prefiltered_project_hts(self, project_samples: dict, n_partitions: int, **kwargs):
+        prefiltered_project_hts = []
         project_hts = []
         sample_data = {}
-        for project_guid, project_sample_data in project_samples.items():
-            sample_type = list(project_sample_data.keys())[0]
-            project_ht = self._read_table(f'projects/{sample_type}/{project_guid}.ht')
-
+        for project_guid, project_sample_type_data in project_samples.items():
+            sample_type, family_sample_data = list(project_sample_type_data.items())[0]
+            project_ht = self._read_project_table(project_guid, sample_type)
             if project_ht is None:
                 continue
-            project_hts.append(project_ht.select_globals('sample_type', 'family_guids', 'family_samples'))
-            sample_data.update(project_sample_data[sample_type])
+            project_ht = project_ht.select_globals('sample_type', 'family_guids', 'family_samples')
+            project_hts.append(project_ht)
+            sample_data.update(family_sample_data)
 
-            if len(project_hts) >= chunk_size:
-                self._filter_merged_project_hts(
-                    project_hts, sample_data, filtered_project_hts, filtered_comp_het_project_hts, n_partitions, **kwargs,
+            if len(project_hts) >= HT_CHUNK_SIZE:
+                self._prefilter_merged_project_hts(
+                    project_hts, sample_data, prefiltered_project_hts, n_partitions, **kwargs
                 )
                 project_hts = []
                 sample_data = {}
 
-        self._filter_merged_project_hts(
-            project_hts, sample_data, filtered_project_hts, filtered_comp_het_project_hts, n_partitions, **kwargs,
+        self._prefilter_merged_project_hts(
```

Comment:
The intention was keep the flow close to what's happening now for a single sample type, and everything up to applying the prefilters was the same, but applying quality and inheritance filters is where we needed to start special handling for both sample types. 

I agree that it can be confusing, so I will remove 'prefilter' from most of functions that have prefilter in the name, so `_load_prefiltered_project_hts` becomes `_load_project_hts`. However `_prefilter_merged_project_hts` is the only fn for which I'll maintain the prefilter rename.